src/compiler/checker.ts

```diff
@@ -6800,15 +6962,16 @@ module ts {
                     let paramType = getTypeAtPosition(signature, i);
                     // A tagged template expression provides a special first argument, and string literals get string literal types
                     // unless we're reporting errors
-                    let argType = i === 0 && node.kind === SyntaxKind.TaggedTemplateExpression
-                        ? globalTemplateStringsArrayType
-                        : arg.kind === SyntaxKind.StringLiteral && !reportErrors
+                    let argType = getSyntheticArgumentType(i, arg);

```

Comment:
I say this because unless this gets inlined (I can't imagine it will, the function is quite large), we'll be calling this for every single argument in a call.